sympytorch/sympy_module.py

```diff
@@ -104,10 +108,12 @@ def __init__(self, *, expr, _memodict, _func_lookup, **kwargs):
     def sympy(self, _memodict):
         if issubclass(self._sympy_func, sympy.Float):
             return self._sympy_func(self._value.item())
-        elif issubclass(self._sympy_func, sympy.UnevaluatedExpr):
-            return self._sympy_func(self._value.item())
         elif issubclass(self._sympy_func, sympy.Integer):
             return self._sympy_func(self._value)
+        elif issubclass(self._sympy_func, sympy.core.numbers.Half) or issubclass(self._sympy_func, sympy.core.numbers.Rational) :
+            return self._sympy_func(self._value)    
```

Comment:
I think this will fail for `Half`:
```python
>>> sympy.core.numbers.Half(0.5)`
# TypeError: Singleton.__init__.<locals>.<lambda>() takes 1 positional argument but 2 were given
```